pkg/clusterresource/builder.go

```diff
@@ -157,8 +160,8 @@ func (o *Builder) Build() ([]runtime.Object, error) {
 	var allObjects []runtime.Object
 	allObjects = append(allObjects, o.generateClusterDeployment())
 
-	if mp := o.generateMachinePool(); mp != nil {
-		allObjects = append(allObjects, mp)
+	if !o.SkipMachinePools {
+		allObjects = append(allObjects, o.generateMachinePool())
```

Comment:
Please keep the nil check to be able to support platforms that do not support MachinePools.